app/create-app-template.js

```diff
@@ -104,6 +103,24 @@ module.exports = api => {
       )
       .join('\n')}
 
+    if (entry.extendRootOptions) {
+      entry.extendRootOptions(rootOptions)
+    }
+
+    const { router } = rootOptions
+
+    if (__DEV__) {
+      if (!router) {
+        throw new Error(${JSON.stringify(
+          `You must ${
+            api.options.fsRoutes
+              ? `create ${api.options.fsRoutes.path}/*.vue or `
+              : ``
+          }export "router" in ${api.options.entry || `entry file`}!`
+        )})
+      }
+    }
```

Comment:
I'm thinking about create a default `router` instance here `entry.router` is not provided, and use `router.addRoutes` to add routes in the `fs-routes` plugin.

Reply:
I reworked it the way you described. One of the benefits it's that it's now possible to merge entry routes with fs routes (with former taking precedence).

However, it's now not possible to detect a misconfigured router in `create-app-template.js`, as router doesn't expose the full set of routes (you can't rely on `router.options.routes` as it only contains the initial set of routes and not the ones added by `addRoutes`):
https://github.com/vuejs/vue-router/blob/4cb6699ecc4453a992429b28f32f4add3e2680b6/src/create-matcher.js#L171

Thoughts?